packages/storybook-addon/src/lib/storybook-addon.ts

```diff
@@ -0,0 +1,156 @@
+  const webpackVersion = await presets.apply('webpackVersion');
+  logger.info(`=> [MF] Webpack ${webpackVersion} version detected`);
+
+  if (webpackVersion !== '5') {
+    throw new Error(
+      'Webpack 5 required: Configure Storybook to use the webpack5 builder'
+    );
+  }
+
+  if (nxModuleFederationConfig) {
+    logger.info(`=> [MF] Detect NX configuration`);
+
+    const wmf = await withModuleFederation(nxModuleFederationConfig);
+
+    webpackConfig = {
+      ...webpackConfig,
+      ...wmf(webpackConfig),
+    };
+  }
+
+  if (moduleFederationConfig) {
+    logger.info(`=> [MF] Push Module Federation plugin`);
+    plugins.push(new ModuleFederationPlugin(moduleFederationConfig));
+  }
+
+  const entries = await presets.apply<string[]>('entries');
+  const bootstrap: string[] = entries.map(
+    (entryFile: string) => `import '${correctImportPath(context, entryFile)}';`
+  );
+
+  const index = plugins.findIndex(
+    (plugin) => plugin.constructor.name === 'VirtualModulesPlugin'
+  );
+
+  if (index !== -1) {
+    logger.info(`=> [MF] Detected plugin VirtualModulesPlugin`);
+
+    /* eslint-disable  @typescript-eslint/no-explicit-any */
+    const plugin = plugins[index] as any;
+
+    const virtualEntries = plugin._staticModules; // TODO: Exist another way to get virtual modules? Or maybe it's good idea to open a PR adding a method to get modules?
+    const virtualEntriesPaths: string[] = Object.keys(virtualEntries);
+
+    logger.info(`=> [MF] Write files from VirtualModulesPlugin`);
+    for (const virtualEntryPath of virtualEntriesPaths) {
+      const nodeModulesPath = '/node_modules/';
+      const filePathFromProjectRootDir = virtualEntryPath.replace(context, '');
+      let sourceCode = virtualEntries[virtualEntryPath];
+      let finalPath = virtualEntryPath;
+      let finalDir = dirname(virtualEntryPath);
+
+      // If virtual file is not in directory node_modules, move file in directory node_modules/.cache/storybook
+      if (!filePathFromProjectRootDir.startsWith(nodeModulesPath)) {
+        finalPath = join(
+          context,
+          nodeModulesPath,
+          '.cache',
+          'storybook',
+          filePathFromProjectRootDir
+        );
+        finalDir = dirname(finalPath);
+
+        // Fix storybook stories' path in virtual module `generated-stories-entry.cjs`
+        if (filePathFromProjectRootDir === '/generated-stories-entry.cjs') {
+          const nonNormalizedStories = await presets.apply<string[]>('stories');
+          const stories = normalizeStories(nonNormalizedStories, {
+            configDir: options.configDir,
+            workingDir: context,
+          });
+
+          // For each story fix the import path
+          stories.forEach((story) => {
+            // Go up 3 times because the file was moved in /node_modules/.cache/storybook
+            const newDirectory = join('..', '..', '..', story.directory);
```

Comment:
Why not resolve it from root since we know `node_modules` exists in root?

Reply:
I don't know if I understood the question correctly but it is in the root `node_module` folder. I'm updating the path because the file is written in the folder `/node_modules/.cache/storybook`

Reply:
Ahh okay. My bad.. I have read the code in a wrong way.